src/errors.md

```diff
@@ -0,0 +1,128 @@
+```
+
+Please login again in order to refresh the token.
+
+> Note: Please be aware that initiating a new login will logout any already active connection!
+
+</details>
+
+<details>
+<summary><b>No asset for key <KEY></b></summary>
+
+```sh
+get dataset 09c741742ec4ce10360e0dcd1ca0b8ecf8edf97263593d5f6b1ce6c657d54c8c --profile node-1
+Requests error status 404: {"message":"no asset for key 09c741742ec4ce10360e0dcd1ca0b8ecf8edf97263593d5f6b1ce6c657d54c8c"}
+Error: Request failed: NotFound: 404 Client Error: Not Found for url: http://substra-backend.node-1.com/data_manager/09c741742ec4ce10360e0dcd1ca0b8ecf8edf97263593d5f6b1ce6c657d54c8c/
+```
+
+Please check the item key you are providing (with `substra list <ASSET>`).
+
+</details>
+
+## Python SDK
+
+[SDK documentation](https://github.com/SubstraFoundation/substra/blob/master/references/sdk.md#substrasdk)
+
+> Remember: debug mode is awesome, use it! You just need to define your client like this `client = substra.Client(debug=True)`
+> to be able to use `pdb`/`ipdb`!
+
+### Generate data sample
+
+Check the `generate_data_samples.py` script and the path to your dataset, the values of `N_TRAIN_DATA_SAMPLES` and `N_TEST_DATA_SAMPLES`.
+
+<details>
+<summary><b>FileExistsError</b></summary>
+This indicates that the `train_data_samples` & `test_data_samples` have already been generated in the `assets` folder. This means that you will need to remove it before re-generating data samples.
+</details>
+
+### Register dataset and objective
+
+Check the `add_dataset_objective.py` script and your assets (`DATASET`, `TEST_DATA_SAMPLES_PATHS`, `TRAIN_DATA_SAMPLES_PATHS`, `OBJECTIVE`, `METRICS_DOCKERFILE_FILES`).
+
+### Add an algorithm
+
+Check the `add_algo.py` script and your assets (`ALGO`, `ALGO_DOCKERFILE_FILES`).
+
+<details>
+<summary><b>Status failed</b></summary>
+
+If after adding an algorithm it directly gets a `failed` status, it might be related to the `permissions`: please be sure to allow, in the dataset from the data provider org, the algo provider org to access the related dataset. You will also have to edit the algo permissions and to make it available for the data provider org.
+
+> Note: this is not required in debug mode
+
+</details>
+
+### Add a train tuple
+
+In debug mode, when you run the `add_train_tuple` method with large data, a Docker container is launched and then exit showing this last log `substratools.algo - launching training task` and no further stacktrace. This might be related to a lack of memory (use `docker stats`) and you tweak this limit by using a higher `shm_size` in the `spawner.py` file.
+
+## Exhaustive list of exceptions
+
+If you want to go further, you can have a look to the [full list of exceptions](https://github.com/SubstraFoundation/substra-backend/blob/master/backend/substrapp/tasks/exceptions.json) handled by the backend. This can provide you with some insight about the issue you are facing.
```

Comment:
I would remove this part since the user does not have access to these exceptions or these error codes.

Reply:
It will not be fixed as we are thinking about a new way to display errors and logs.

Reply:
Ok noted